internal/controller/thanosruler_controller.go

```diff
@@ -275,10 +378,16 @@ func (r *ThanosRulerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&appsv1.StatefulSet{}).
 		Owns(&policyv1.PodDisruptionBudget{}).
 		Owns(&monitoringv1.ServiceMonitor{}).
+		Watches(
```

Comment:
here again, I think we might want to simply not add this watch if the flag on the controller isnt enabled